crates/turbopack-core/src/source_map/mod.rs

```diff
@@ -320,6 +478,63 @@ impl SectionedSourceMap {
     pub fn new(sections: Vec<SourceMapSection>) -> Self {
         Self { sections }
     }
+
+    pub async fn source_contents(&self) -> Result<Vec<(String, Option<String>)>> {
+        let mut sources = IndexMap::new();
+
+        for section in &self.sections {
+            let map = section.map.await?;
+            for (source, contents) in map.source_contents().await? {
+                match sources.entry(source) {
```

Comment:
This is just `sources.entry(source).or_insert(contents)`, or really just `sources.insert(source, contents)`?